src/telemetry.c

```diff
@@ -401,6 +401,64 @@ static int set_timestamp_header(struct telem_ref *t_ref)
 }
 
 /**
+ * Sets cpu model for telemetry record. The information from cpu is extracted
+ * from /proc/cpuinfo, specifically "model name" attribute.
+ *
+ * @param t_ref Telemetry Record reference obtained from tm_create_record.
+ *
+ * @return 0 if successful, or a negative errno-style value if not.
+ *
+ */
+static int set_cpu_model_header(struct telem_ref *t_ref)
+{
+        FILE *fs = NULL;
+        char buf[SMALL_LINE_BUF] = { 0 };
+        char *model_name = NULL;
+        const char *attr_name = "model name";
+        int status = 0;
+        size_t attr_len = sizeof(attr_name);
+        size_t size = 0;
+
+        fs = fopen("/proc/cpuinfo", "r");
+        if (fs != NULL) {
+                while (fgets(buf, SMALL_LINE_BUF, fs)) {
+                        if(strncmp(attr_name, buf, attr_len) == 0){
+                            model_name = strchr(buf, ':');
+                            break;
+                        }
+                }
+		fclose(fs);
+		if (model_name != NULL) {
+		        puts("HELLO");	
+                	size = strlen(model_name);
+                	if (size > 2) {
+                    		model_name = (char *) model_name + 2 * sizeof(char);
+                        }
+			size = strlen(model_name);
```

Comment:
This is a nitpick but I would prefer to `size -= 2` in the `if` block above rather than recalculating the string length.